src/main/app/controller/courts/CourtTypesController.ts

```diff
@@ -24,64 +30,79 @@ export class CourtTypesController {
     res: Response,
     updated = false,
     error = '',
-    courtCourtTypes: CourtType[] = null): Promise<void> {
-    if (!courtCourtTypes) {
-
-      const slug: string = req.params.slug as string;
-      await req.scope.cradle.api.getCourtCourtTypes(slug)
-        .then((value: CourtType[]) => courtCourtTypes = value)
-        .catch(() => error += this.getCourtTypesErrorMsg);
-
+    courtTypesAndCodes: CourtTypesAndCodes = null ): Promise<void> {
+    const slug: string = req.params.slug as string;
+    if (courtTypesAndCodes == null) {
+      await req.scope.cradle.api.getCourtTypesAndCodes(slug)
+        .then((value: CourtTypesAndCodes) => courtTypesAndCodes = value)
+        .catch(() => error += this.getCourtTypesAndCodesErrorMsg);
     }
 
-    let allCourtTypes: CourtType[] = [];
+    if(courtTypesAndCodes) {
+      if (!courtTypesAndCodes.dxCodes?.some(ot => ot.isNew === true)) {
+        this.addEmptyFormsForNewEntries(courtTypesAndCodes.dxCodes);
+      }
+    }
 
-    await req.scope.cradle.api.getCourtTypes()
-      .then((value: CourtType[]) => allCourtTypes = value)
+    let courtTypes: CourtType[] = [];
+    await req.scope.cradle.api.getCourtTypes(slug)
+      .then((value: CourtType[]) => courtTypes = value)
       .catch(() => error += this.getCourtTypesErrorMsg);
 
-
     const pageData: CourtTypePageData = {
       errorMsg: error,
       updated: updated,
-      items: this.mapCourtTypeToCourtTypeItem(allCourtTypes, courtCourtTypes)
-
+      items: courtTypesAndCodes && courtTypesAndCodes.types ? this.mapCourtTypeToCourtTypeItem(courtTypes, courtTypesAndCodes.types) : this.mapCourtTypeToCourtTypeItem(courtTypes, []),
```

Comment:
Might be better to call this courtTypes, rather than items? But yeah doesn't matter too much

Reply:
changed to courtTypes to make things bit more easy to follow